src/bare-metal/alloc.md

```diff
@@ -0,0 +1,16 @@
+# `alloc`
+
+To use `alloc` you must implement a global (heap) allocator.
+
+```rust,editable,compile_fail
+{{#include alloc-example/src/main.rs:Alloc}}
+```
+
+<details>
+
+* `buddy_system_allocator` is a third-party crate implementing a basic buddy system allocator. Other
+  crates are available, or you can write your own or hook into your existing allocator.
+* The const parameter of `LockedHeap` is the max order of the allocator; i.e. in this case it can
+  allocate regions of up to 2**32 bytes.
+
```

Comment:
Questions that come to mind:

* What are the differences from this program and a "normal" program without `no_std`?
* Can I use all heap allocating types like normal? 
* Do third-party libraries need to do anything to use the global allocator? (I suppose not since it's, well, global).

Reply:
The difference is that you can use the `alloc` crate. The `no_std.md` page has an overview of what that includes. It doesn't include all heap allocating types; `HashMap` is only in `std` for example, but it includes most of them.

Reply:
Okay, thanks.